src/dialects/postgres/query-generator-typescript.ts

```diff
@@ -48,4 +51,34 @@ export class PostgresQueryGeneratorTypeScript extends AbstractQueryGenerator {
       'GROUP BY i.relname, ix.indexrelid, ix.indisprimary, ix.indisunique, ix.indkey ORDER BY i.relname;',
     ]);
   }
+
+  removeIndexQuery(tableName: TableNameOrModel, indexNameOrAttributes: string | string[], options: RemoveIndexQueryOptions) {
+    if (options) {
+      rejectInvalidOptions(
+        'removeIndexQuery',
+        this.dialect.name,
+        REMOVE_INDEX_QUERY_SUPPORTABLE_OPTIONS,
+        new Set(['concurrently', 'ifExists', 'cascade']),
+        options,
+      );
+    }
+
+    let indexName;
+    const table = this.extractTableDetails(tableName);
+    if (Array.isArray(indexNameOrAttributes)) {
+      indexName = generateIndexName(table, { fields: indexNameOrAttributes });
+    } else {
+      indexName = indexNameOrAttributes;
+    }
+
+    return joinSQLFragments([
+      'DROP INDEX',
+      options?.concurrently ? 'CONCURRENTLY' : '',
+      options?.ifExists ? 'IF EXISTS' : '',
+      table?.schema && table.schema !== this.dialect.getDefaultSchema()
+        ? `${this.quoteIdentifier(table.schema)}.${this.quoteIdentifier(indexName)}`
+        : this.quoteIdentifier(indexName),
+      options?.cascade && !options?.concurrently ? 'CASCADE' : '',
```

Comment:
Can we throw an error if both `cascade` and `concurrently` is used? So the user knows that the two options can't be combined?